File: app/src/main/java/com/machinelearning/playcarddetect/modules/accessibilityaction/action/SwipeAction.java
package com.machinelearning.playcarddetect.modules.accessibilityaction.action;


import android.graphics.RectF;

public class SwipeAction extends GestureAction {
    RectF swipeStartRectF = new RectF();
    RectF swipeEndRectF = new RectF();
    public SwipeAction(RectF swipeStartRectF,RectF swipeEndRectF, long startTime, long durationTime, int delayTime) {
        super(startTime,durationTime,delayTime,"Swipe");
        this.swipeStartRectF =swipeStartRectF;
        this.swipeEndRectF = swipeEndRectF;
    }

    public SwipeAction() {
    }

    public long getDurationTime() {
        return durationTime;
    }

    public void setDurationTime(long durationTime) {
        this.durationTime = durationTime;
    }

    public long getStartTime() {
        return startTime;
    }

    public void setStartTime(long startTime) {
        this.startTime = startTime;
    }

    public RectF getSwipeStartRectF() {
        return swipeStartRectF;
    }

    public void setSwipeStartRectF(RectF swipeStartRectF) {
        this.swipeStartRectF = swipeStartRectF;
    }

    public RectF getSwipeEndRectF() {
        return swipeEndRectF;
    }

    public void setSwipeEndRectF(RectF swipeEndRectF) {
        this.swipeEndRectF = swipeEndRectF;
    }
}

File: app/src/main/java/com/machinelearning/playcarddetect/modules/accessibilityaction/action/SwipeAction.kt
package com.nhatran241.accessibilityactionmodule.model

import android.accessibilityservice.GestureDescription
import android.accessibilityservice.GestureDescription.StrokeDescription
import android.graphics.Path
import android.graphics.RectF
import androidx.annotation.NonNull
import androidx.annotation.Nullable
import com.machinelearning.playcarddetect.modules.accessibilityaction.action.GestureAction

class SwipeAction(@NonNull startRectF: RectF, @NonNull endRectF: RectF, @Nullable startTime:Long, @Nullable duration: Long, @Nullable delayTime : Long) : GestureAction(){
    val isSwipe =true
    init {
        val path = Path()
        this.delayTime =delayTime
        val builder = GestureDescription.Builder()
        path.moveTo(startRectF.centerX(),startRectF.centerY())
        path.lineTo(endRectF.centerX(),endRectF.centerY())
        builder.addStroke(StrokeDescription(path, startTime, duration))
        gestureDescription=builder.build()
    }
}